src/utils.ts

```diff
@@ -179,3 +179,12 @@ export function getSnapPoint(map: mapgl.Map, joints: Joint[], point: ScreenPoint
 
     return { point: geoPoint, distance, segment: bestSegmentIndex };
 }
+
+export function getMousePosition(
+    container: HTMLElement,
+    clientX: number,
+    clientY: number,
+): number[] {
+    const rect = container.getBoundingClientRect();
```

Comment:
NOTE: all the screencasts above have another the constant displacement issue. Pay attention only to the movement of the ruler joints.

The mouse position issue is still persist if we change additional CSS as
```diff
  #container {
+     padding: 30px;
-     position: absolute;
-     top: 30px;
-     left: 30px;
-     margin: 0;
-     width: calc(100% - 60px);
-     height: calc(100% - 60px);
-     overflow: hidden;
  }
```
CSS change result:

https://user-images.githubusercontent.com/30019338/193046760-5f17bdfb-3699-42dd-9ff5-a159d8bae22c.mp4

Here we have to base on the nested `canvas` element:
```js
// Or we can use here some `children` logic
const rect = container.querySelector('canvas')!.getBoundingClientRect();
```
CSS + rect changes result:

https://user-images.githubusercontent.com/30019338/193047806-43b182ee-86f3-428e-9460-5758c4a88e67.mp4